Code cleanup and api function format for Group

atelic opened the pull request:
All of the `POST`, `PATCH`, and `DELETE` calls to the google groups api do not throw errors. They also all use the service method instead of the raw http request. These changes are still not showing in the admin
console. This needs to be investigated.

Hypothetically, any authorized, logged in superuser should be able to create, update, and destroy a google group and mailing list with the way the code is currently implemented

overstreetce commented:
When you add to the commit, can you get rid of the .idea file? We need to remove those and it won't merge until you remove it as well

atelic committed:
Code cleanup and api function format for Groups
All of the POST, PATCH, and DELETE calls to the google groups
api do not throw errors. They also all use the service method instead of
the raw http request. These changes are still not showing in the admin
console. This needs to be investigated.

atelic committed:
Fix: was not calling `super` for delete and update
